plugins/modules/custom_image.py

```diff
@@ -0,0 +1,479 @@
+  url:
+    description:
+      - The URL used to download the image.
+    type: str
+  name:
+    description:
+      - The human readable name of the custom image. Either name or UUID must
+        be present to change an existing image.
+    type: str
+  uuid:
+    description:
+      - The unique identifier of the custom image import. Either name or UUID
+        must be present to change an existing image
+    type: str
+  slug:
+    description:
+      - A string identifying the custom image for use within the API.
+    type: str
+  user_data_handling:
+    description:
+      - How user_data will be handled when creating a server. There are
+        currently two options, "pass-through" and "extend-cloud-config".
+    type: str
+    choices: [ pass-through , extend-cloud-config ]
+  zones:
+    description:
+      - Specify zones in which the custom image will be available (e.g. C(lpg1)
+        or C(rma1)).
+    type: list
+    elements: str
+  source_format:
+    description:
+      - The file format of the image referenced in the url. Currently only raw
+        is supported.
+    choices: [ raw ]
+    type: str
+  tags:
+    description:
+      - The tags assigned to the custom image.
+    type: dict
+  state:
+    description: State of the coustom image.
+    choices: [ present, absent ]
+    default: present
+    type: str
+extends_documentation_fragment: cloudscale_ch.cloud.api_parameters
+'''
+
+EXAMPLES = r'''
+- name: Import custom image
+  cloudscale_ch.cloud.custom_image:
+    name: "My Custom Image"
+    url: https://ubuntu.com/downloads/hirsute.img
+    slug: my-custom-image
+    user_data_handling: extend-cloud-config
+    zones: lpg1
+    tags:
+      project: luna
+    state: present
+  register: my_custom_image
+
+- name: Wait until import succeeded
+  cloudscale_ch.cloud.custom_image:
+    uuid: "{{ my_custom_image.uuid }}"
+  retries: 15
+  delay: 5
+  register: image
+  until: image.import_status == 'success'
+  failed_when: image.import_status == 'failed'
+
+- name: Import custom image and wait until import succeeded
+  cloudscale_ch.cloud.custom_image:
+    name: "My Custom Image"
+    url: https://ubuntu.com/downloads/hirsute.img
+    slug: my-custom-image
+    user_data_handling: extend-cloud-config
+    zones: lpg1
+    tags:
+      project: luna
+    state: present
+  retries: 15
+  delay: 5
+  register: image
+  until: image.import_status == 'success'
+  failed_when: image.import_status == 'failed'
+
+- name: Update custom image
+  cloudscale_ch.cloud.custom_image:
+    name: "My Custom Image"
+    slug: my-custom-image
+    user_data_handling: extend-cloud-config
+    tags:
+      project: luna
+    state: present
+
+- name: Delete custom image
+  cloudscale_ch.cloud.custom_image:
+    uuid: '{{ my_custom_image.uuid }}'
+    state: absent
+
+- name: List all custom images
+  uri:
+    url: 'https://api.cloudscale.ch/v1/custom-images'
+    headers:
+      Authorization: 'Bearer {{ cloudscale_api_token }}'
```

Comment:
While this works, I suggest to use the canonical way of providing the token for the Ansible modules:
```suggestion
      Authorization: 'Bearer {{ query("env", "CLOUDSCALE_API_TOKEN") }}'
```